infrastructure/src/main/java/org/corfudb/infrastructure/BatchWriter.java

```diff
@@ -66,6 +66,21 @@ public void write(@Nonnull K key, @Nonnull V value) {
         }
     }
 
+    public void bulkWrite(List<LogData> entries) {
+        try {
+            CompletableFuture<Void> cf = new CompletableFuture();
+            operationsQueue.add(new BatchWriterOperation(BatchWriterOperation.Type.BULK_WRITE,
+                    null,null, entries, cf));
+        } catch (Exception e) {
+            log.trace("Write Exception {}", e);
+            if (e.getCause() instanceof RuntimeException) {
+                throw (RuntimeException)e.getCause();
+            } else {
+                throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)